pkg/scheduler/framework/plugins/noderesources/node_resources.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesources
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/algorithm/predicates"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/migration"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	"k8s.io/kubernetes/pkg/scheduler/nodeinfo"
+)
+
+// NodeResources is a plugin that checks if a node has sufficient resources.
+type NodeResources struct{}
+
+var _ = framework.FilterPlugin(&NodeResources{})
+
+// Name is the name of the plugin used in the plugin registry and configurations.
+const Name = "NodeResources"
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (pl *NodeResources) Name() string {
+	return Name
+}
+
+// Filter invoked at the filter extension point.
+func (pl *NodeResources) Filter(_ *framework.CycleState, pod *v1.Pod, nodeInfo *nodeinfo.NodeInfo) *framework.Status {
+	_, reasons, err := predicates.PodFitsResources(pod, nil, nodeInfo)
```

Comment:
We need to read the metadata from CycleState and pass it to PodFitsResources, it is used inside the function.

Reply:
done